### DISCUSSION TITLE
Scalar product of a Four vector and a polarization vector

### DISCUSSION THREAD

**Discussion by Joro20** [FeynCalc/feyncalc#268] [Q&A]
Hello,

I'm very new to the world of FeynCalc. 
I'm trying to calculate the W^+W^- scattering process. As it consists of 7 Feynman Diagrams, I was just trying to calculate the amplitude of the s-channel diagram with photon mediator. So, I generated this matrix element:

 `PolarizationVector[p1, \[Alpha]] PolarizationVector[
  p2, \[Beta]] (MT[\[Alpha], \[Beta]] FV[p1 - p2, \[Mu]] + 
    MT[\[Beta], \[Mu]] FV[2 p2 + p1, \[Alpha]] + 
    MT[\[Mu], \[Alpha]] FV[-2 p1 - p2, \[Beta]]).(-((
    I MT[\[Mu], \[Nu]])/
    SP[k])).(MT[\[Nu], \[Gamma]] FV[2 q1 + q2, \[Delta]] + 
    MT[\[Gamma], \[Delta]] FV[q2 - q1, \[Nu]] + 
    MT[\[Delta], \[Nu]] FV[-q1 - 2 q2, \[Gamma]]) ComplexConjugate[
  PolarizationVector[q1, \[Gamma]]] ComplexConjugate[
  PolarizationVector[q2, \[Delta]]]`

Now, when I'm trying to calculate this, I'm facing a couple of issues which are the following:
a) How to provide specific values/components to a Four-vectors, as well as polarization vectors? 
b) How to perform a Scalar product of a Four-vector and a polarization vector with such specified components? Several such terms appear after I contract the above matrix element.

Please help.

**Comment by vsht** [comment]
Hi,

this is not something you can directly do in FeynCalc. The package 
returns you a result written in terms of Lorentz invariant quantities,
e.g. scalar products of 4-vectors. Then you are free to substitute those
with whatever you want.

However, you cannot specify single components from the very beginning
and do the calculation in this fashion.

Cheers,
Vladyslav

Am 05.05.24 um 15:59 schrieb Joro20:

**Reply by Joro20** [reply]
Hi Vladyslav,

Thanks for your reply. Since both of my concerns can't be implemented, that precisely means, the WW scattering calculations can't be done in FeynCalc in a straightforward manner! In that case, can you please suggest any alternative for such calculation ? Because, even if the s-channel calculations can be taken care of , t-channel processes become tedious while do them by hand!

Thanks,
Joydeep

**Reply by anaik49** [reply]
Hi,
I am sort of dealing with a similar problem. I can define all the scalar products of the 4-vectors and substitute those in, but I cannot define the contraction of those 4-vectors with the Levi-Civita tensor. To do that, I have to do the contraction over the indices with an explicit sum using the mathematica LeviCivitaTensor function, but still cannot substitute that expression in for the Levi-Civita term that FeynCalc produces. Is there a work around to this?

Thanks,
Atharva

**Reply by anaik49** [reply]
Nevermind I figured it out, thank you!
